### PR TITLE
Add kubectl diff exit code doc

### PR DESCRIPTION
**Pull request by zhouya0**
**What type of PR is this?**
> /kind documentation

**What this PR does / why we need it**:
As this PR https://github.com/kubernetes/kubernetes/pull/87437 being merged, add `kubectl diff` exit code documentation to indicates that the `KUBECTL_EXTERNAL_DIFF` programme must use 1 as changes detected.

**Which issue(s) this PR fixes**:

Fixes #https://github.com/kubernetes/kubernetes/issues/87343

/assign @apelisse

### CODE REVIEW

File: staging/src/k8s.io/kubectl/docs/book/pages/app_composition_and_deployment/diffing_local_and_remote_resources.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ The `KUBECTL_EXTERNAL_DIFF` environment variable can be used to select your own
 By default, the "diff" command available in your path will be run with "-u" (unified) and "-N"
 (treat new files as empty) options.
 
+If the exit code after diff command is 0, means there is no change detected and the exit code 1 means there is at least one change which indicates `KUBECTL_EXTERNAL_DIFF` programs must also follow the convention of using 1 as their exit code for changes detected. If the exit code is more than 1, means error occurs in this command.
```
Comment (apelisse) [inline staging/src/k8s.io/kubectl/docs/book/pages/app_composition_and_deployment/diffing_local_and_remote_resources.md]: Some of diff man-pages are presented like this:
```
EXIT STATUS
The following exit values shall be returned:

 0
No differences were found.
 1
Differences were found.
>1
An error occurred.
```

which seems much simpler.

Here's my suggestion:
```
EXIT STATUS
The following exit values shall be returned:

 0
No differences were found.
 1
Differences were found.
>1
Kubectl or diff failed with an error.

Note: `KUBECTL_EXTERNAL_DIFF`, if used, is expected to follow that convention.
```

What do you think?

Comment (apelisse) [reply staging/src/k8s.io/kubectl/docs/book/pages/app_composition_and_deployment/diffing_local_and_remote_resources.md]: We might want to document this in the kubectl command directly too, thanks!

Comment (zhouya0) [reply staging/src/k8s.io/kubectl/docs/book/pages/app_composition_and_deployment/diffing_local_and_remote_resources.md]: Good, thanks! I made changes.

File: staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go [hunk 2]
```diff
@@ -56,7 +56,13 @@ var (
 
 		KUBECTL_EXTERNAL_DIFF environment variable can be used to select your own
 		diff command. By default, the "diff" command available in your path will be
-		run with "-u" (unified diff) and "-N" (treat absent files as empty) options.`))
+		run with "-u" (unified diff) and "-N" (treat absent files as empty) options.
```
Comment (apelisse) [inline staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go]: Actually, can you use the same format as above? I think it's clearer that the long sentence. Thanks!

Comment (apelisse) [reply staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go]: Wait no this part is good, you can keep it as it is.

File: staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go [hunk 3]
```diff
@@ -56,7 +56,13 @@ var (
...
+		
+		If the exit code after diff command is 0, means there is no change detected
```
Comment (apelisse) [inline staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go]: This is the part that you should replace.

Comment (zhouya0) [reply staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go]: Ok, done!

**Comment by apelisse** [comment]
This shouldn't have broken the tests?
/retest
/lgtm
/approve

**Comment by apelisse** [comment]
/kind documentation

@zhouya0 Please, pay more attention to the template next time :-)

**Pull request merged**